### ISSUE TITLE
Unexpected error imposed by expression MAX(null) < LENGTH('rh')

### ISSUE THREAD

**Issue by sayJason** [cockroachdb/cockroach#99007]
**Describe the problem**

Please describe the issue you observed, and any steps we can take to reproduce it:

**To Reproduce**
Run the following statements.
```SQL
SELECT MAX(null) < LENGTH('rh');
```

**Expected behavior**
Return value NULL, instead of reporting errors, in which the error message is confusing.
```SQL
ERROR: unknown signature: length(string) (desired <string>)
SQLSTATE: 42883
```

**Environment:**
 - CockroachDB version 22.2.5
 - Server OS: Ubuntu 20.04LTS
 - Client app: cockroach sql

Jira issue: CRDB-25652

**Comment by chengxiong-ruan** [comment]
Thanks for filing the issue. Looks like this has been fixed on master and the coming new versions, and it now returns a better error message :
```
ERROR: unsupported comparison operator: <string> < <int>
SQLSTATE: 22023
```

In 22.2, the error message was a bit confusing, but functionally I think it make sense. Because `MAX(NULL)` was type checked as a string return type, and we were trying to find a `length()` function which returns a string type which does not exist. But agree that we should return a better error message than this.

**Comment by msirek** [comment]
Looks like @chengxiong-ruan found this is already fixed in 23.1.
Closing.
